### DISCUSSION TITLE
tailwindcss + next.js + emotion.js setup

### DISCUSSION THREAD

**Discussion by theairbend3r** [tailwindlabs/tailwindcss#1474] [General]
Hi!

First off, I love the tailwind library! 

I love the tailwindcss + nextjs setup given in [tailwindcss/setup-examples](https://github.com/tailwindcss/setup-examples/tree/master/examples/nextjs).

I was wondering if that example can be extended to use emotion.js too (with explanations)?

Let me know your thoughts! :)

**Comment by merelinguist** [comment]
https://codesandbox.io/s/youthful-gauss-cjfsf

Does this help?

**Comment by theairbend3r** [comment]
Hi Dylan,

Thanks for the quick reply! This works well. What extra things did you do, after following the next.js example in [tailwindcss/setup-examples](https://github.com/tailwindcss/setup-examples/tree/master/examples/nextjs)?

I have one more question, how do use tailwind to create components as well? In the codesandbox you shared above, a component is created using CSS - 
```
const Root = styled.div`
  max-width: 24rem;
  margin: 0 auto;
`;
```

How can I use tailwind to create a component? Something like the following -
```
const Root = styled.div`
  font-bold text-red-500
`;

<Root> Hello World </Root>
```

Next.js' Github repo has an example where we use the `tailwind.macro` to achieve this (but I think it's no up to date). Check it out [here](https://github.com/zeit/next.js/tree/canary/examples/with-tailwindcss-emotion).

As an aside, I think [tailwind.macro](https://github.com/bradlc/babel-plugin-tailwind-components) is not actively maintained anymore. [twin.macro](https://github.com/ben-rogerson/twin.macro) is a fork which supports tailwind v1.2. Check it out.
